stdlib/3/itertools.pyi

```diff
@@ -44,20 +44,18 @@ def islice(iterable: Iterable[_T], stop: int) -> Iterator[_T]: ...
 def islice(iterable: Iterable[_T], start: int, stop: Optional[int],
            step: int = ...) -> Iterator[_T]: ...
 
-def starmap(func: Any, iterable: Iterable[Any]) -> Iterator[Any]: ...
+def starmap(func: Callable[..., _S], iterable: Iterable[Iterable[Any]]) -> Iterator[_S]: ...
```

Comment:
No, because the elements in the iterable can have different types and they are unpacked as arguments for the callable. So I guess theoretically it would have to be something like this:

```
def starmap(func: Callable[_T, _S], iterable: Iterable[_T]) -> Iterator[_S]: ...
```

But that is not possible. As an example, `itertools.starmap(lambda x, y: x * len(y), [(5, 'abc'), (2, 'xy')])` would be completely valid, but is impossible to fully generically type with the current system.